### ISSUE TITLE
.bashrc environment variables not set up

### ISSUE THREAD

**Issue by josemonsalve2** [jupyterhub/jupyterhub#2011]
Hi, 

I have been trying to set up a jupiterhub server. I wanted to have it be a daemon service and so I used [this instructions](https://github.com/jupyterhub/jupyterhub/wiki/Run-jupyterhub-as-a-system-service). 

I have successfully spawn it and log in, but I am having issues with the user's environmental variables that are set up in the .bashrc file.

In [this issue](https://github.com/jupyterhub/jupyterhub/issues/948) I saw that it is necessary to modify the `jupiterhub_config.py` with this option: 

```
c.LocalProcessSpawner.shell_cmd = ['bash', '-l', '-c']
```

However, this still does not seem to work. I have set up the user's PATH and LD_LIBRARY_PATH in the `.bashrc` file. But it does not get reflected in the spawned jupyter notebook. 

I appreciate any help you can provide and let me know what other information might be useful for you to help me out.

**Comment by josemonsalve2** [comment]
I have also tried 
```
c.Spawner.shell_cmd = ['bash', '-l', '-c']
``` 

with no luck

**Comment by chicocvenancio** [comment]
Try `c.LocalProcessSpawner.cmd` to get the "shell loading .bashrc route", but I would prefer using `c.LocalProcessSpawner.environment` to set the `PATH` and `LD_LIBRARY_PATH` directly.

**Comment by josemonsalve2** [comment]
@chicocvenancio Thanks for your quick response. 

I think this would work for some cases, but my users (which are trusted users) might have multiple other parameters to be set up by them.

When you say try `c.LocalProcessSpawner.cmd`. Should I set it to:

```
c.LocalProcessSpawner.cmd = ['bash', '-l', '-c']
```

I tried that, but it fails to start the server: 500 Internal error

**Comment by josemonsalve2** [comment]
It seems to be a bug somewhere:

I enabled debugging with: 

```
> sudo /usr/bin/python3 /usr/local/bin/jupyterhub --config=/etc/jupyterhub/jupyterhub_config.py --debug
```

And I realized that there is an error with that option for the LocalProcessSpawner class

```
[W 2018-06-25 16:10:22.039 JupyterHub configurable:168] Config option `shell_cmd` not recognized by `LocalProcessSpawner`.
```

Which occurs because it is not being `.tag(config=True)` in the [spawner.py:1005](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/spawner.py) file. 

However, after fixing it, I see some env variables but not all.

**Comment by josemonsalve2** [comment]
my `~/.bashrc` does not get executed, but '~/.profile` does. This after fixing that bug

**Comment by willingc** [comment]
@josemonsalve2, did you resolve your issue? I believe that you need to `source` your `.bashrc` file from `.profile` for it to execute.

**Comment by willingc** [comment]
Closing this issue since it appears to be resolved.  Thanks to all!

**Comment by fionnmk** [comment]
I still have this issue after following the advice above, can anyone help?


The jupyter Terminal opens in bash, but does not source anything from ~/.profile or ~/.bashrc.

**Comment by jabbera** [comment]
I'm having this issue as well @fionnmk any luck figuring it out?

**Comment by yankee14** [comment]
Had this issue as well.

I debugged by inserting `echo` statements into both `~/.profile` and `~/.bashrc`. When opening a terminal in Jupyter, I see both `echo` statements fire. So they are getting sourced by Jupyter without having to modify either, at least at the time of writing.

Replaced `echo` statements with env var assignment I needed for my use case, which was `export HTTPS_PROXY=http://example.com:12345`. With this, the variable gets sourced, and I am now able to install pip packages.

The environment variable option in `jupyter_config.py` which was mentioned elsewhere in this thread did not work.
